news/2025/2025-08-25-performance-points-star-rating-updates.md

```diff
@@ -0,0 +1,238 @@
+
+Prior to this fix, objects placed 1-5ms off of their intended timing would be treated as entirely new rhythms. These patterns would be played identically, however the system incorrectly saw these as constant, complex rhythm changes and would inflate star rating as a result. In some extreme cases, this caused inflation upwards of 2*.
+
+This fix works by normalising times between objects within a certain tolerance, meaning that 1ms unsnaps will be normalised to the same time and treated as the same rhythm. For maps not abusing this problem, this change has minimal effects.
+
+### Colour penalty fix
+
+A [change](https://github.com/ppy/osu/pull/33641) by [ltca](https://osu.ppy.sh/users/11475208) was made in order to fix a that caused patterns to unfairly avoid colour penalties.
+
+Colour difficulty contains a penalty for long patterns with no rhythm changes, but this penalty would only apply if the pattern's rhythm was *exactly* the same. This was addressed by looking back at previous notes and calculating an average, so that slightly missnapped notes are still correctly penalised.
+
+### Stamina length bonus fix
+
+A [change](https://github.com/ppy/osu/pull/33380) by [rloseise](https://osu.ppy.sh/users/6793778) was made in order to fix high-end stamina values causing a harsh jump in length bonus.
+
+For very difficult maps with a lot of stamina difficulty, the length bonus would previously spike by about 0.2* when reaching a specific threshold value. Length bonus has been rewritten to scale linearly as stamina difficulty increases to prevent harsh increases.
+
+### Minor changes
+
+- A [fix](https://github.com/ppy/osu/pull/33635) by [ltca](https://osu.ppy.sh/users/11475208) to ensure the mono length bonus cannot cause an increase in SR when removing notes
+- A [fix](https://github.com/ppy/osu/pull/34829) by [tsunyoku](https://osu.ppy.sh/users/11315329) to ensure 0-strain maps cannot crash difficulty calculations
+- A [change](https://github.com/ppy/osu/pull/34089) by [rloseise](https://osu.ppy.sh/users/6793778) to reduce the PP bonus for HD on scores set in osu!(lazer) due to its wider aspect ratio making fast reading easier
+- A [change](https://github.com/ppy/osu/pull/33415) by [rloseise](https://osu.ppy.sh/users/6793778) to remove static nerfs for the EZ mod
+- A [change](https://github.com/ppy/osu/pull/33500) by [rloseise](https://osu.ppy.sh/users/6793778) to change how difficulty and accuracy PP are summed
+- A [change](https://github.com/ppy/osu/pull/34962) by [rloseise](https://osu.ppy.sh/users/6793778) to rebalance final values to meet community expectations 
+- A [set](https://github.com/ppy/osu/pull/33233) of [changes](https://github.com/ppy/osu/pull/34327) by [ltca](https://osu.ppy.sh/users/11475208) and [rloseise](https://osu.ppy.sh/users/6793778) to add the consistency factor that is used above
```

Comment:
the what that is used where

Reply:
yeah, this one is a bit more of a technical detail as to how the length bonus and miss penalty actually know what a "consistent" beatmap is :sweat_smile:

tried something in f9758ba916a991115eca88177399e4a2ea2a3415, hopefully this is good